internal/tool/files_by_language.go

```diff
@@ -0,0 +1,166 @@
+	".ts":         "typescript",
+	".tsx":        "typescript",
+	".dockerfile": "dockerfile",
+	"Dockerfile":  "dockerfile",
+	// ".sql"
+	// ".tsql"
+	// ".trg", ".prc", ".fnc", ".pld", ".pls", ".plh", ".plb", ".pck", ".pks", ".pkh", ".pkb", ".typ", ".tyb", ".tps", ".tpb"
+	".json": "json",
+	// ".scss"
+	// ".less"
+	".go": "go",
+	// ".jsp"
+	// ".vm"
+	".xml":     "xml",
+	".xsl":     "xml",  // missing from tests
+	".wsdl":    "xml",  // missing from tests
+	".pom":     "xml",  // missing from tests
+	".cls":     "apex", // missing from tests
+	".trigger": "apex", // missing from testss
+	// ".component", ".page"
+	".cs":  "csharp",
+	".kt":  "kotlin",
+	".kts": "kotlin", // missing from tests
+	".ex":  "elixir", // missing from tests
+	".exs": "elixir",
+	// ".md", ".markdown", ".mdown", ".mkdn", ".mkd", ".mdwn", ".mkdown", ".ron"
+	// ".ps1", ".psc1", ".psd1", ".psm1", ".ps1xml", ".pssc", ".cdxml", ".clixml"
+	// ".cr"
+	// ".cbl", ".cob"
+	// ".groovy"
+	// ".abap"
+	// ".vb"
+	// ".m"
+	".yaml": "yaml", // should these be Terraform?
+	".yml":  "yaml",
+	".dart": "dart", // missing from tests
+	".rs":   "rust",
+	".rlib": "rust", // missing from tests
+	".clj":  "clojure",
+	".cljs": "clojure", // missing from tests
+	".cljc": "clojure", // missing from tests
+	".edn":  "clojure", // missing from tests
+	// ".hs", ".lhs"
+	// ".erl"
+	// ".elm"
+	".html": "html", // missing from tests
+	// ".pl"
+	// ".fs"
+	// ".f90", ".f95", ".f03"
+	".r": "r", // missing from tests
+	// ".scratch", ".sb", ".sprite", ".sb2", ".sprite2"
+	".lua":  "lua",  // missing from tests
+	".asd":  "lisp", // missing from tests
+	".el":   "lisp", // missing from tests
+	".lsp":  "lisp", // missing from tests
+	".lisp": "lisp", // missing from tests
+	// ".P", ".swipl"
+	".jl": "julia", // missing from tests
+	// ".ml", ".mli", ".mly", ".mll"
+	".sol": "solidity",
+	".tf":  "terraform",
+}
+
+func populateFilesByLanguage(toolExecutionFiles *[]string, toolExecutionSourceDir string) error {
+	// If there are files to analyse, analyse only those files
+	if toolExecutionFiles != nil && len(*toolExecutionFiles) > 0 {
+		return populateFilesByLanguageFromFiles(*toolExecutionFiles)
+	}
+	// If there are no files to analyse, analyse all files from source dir
+	return populateFilesByLanguageFromSourceDir(toolExecutionSourceDir)
+}
+
+func populateFilesByLanguageFromFiles(toolExecutionFiles []string) error {
+	for _, file := range toolExecutionFiles {
+		addFileToFilesByLanguage(file)
+	}
+
+	return nil
+}
+
+func populateFilesByLanguageFromSourceDir(toolExecutionSourceDir string) error {
+	// Semgrep can analyse full directories and its subdirectories
+	// but we will have to analyse every extension from every file
+	// so we will have to do this walk somewhere else if we dont do it here
+	err := filepath.WalkDir(toolExecutionSourceDir, walkDirFunc)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func walkDirFunc(path string, info fs.DirEntry, err error) error {
```

Comment:
I think it's best to in-line this function. This name is not providing any information.